src/Features/LanguageServer/ProtocolUnitTests/Workspaces/LspWorkspaceManagerTests.cs

```diff
@@ -116,14 +129,24 @@ public async Task TestLspHasClosedDocumentChangesAsync()
         (_, secondDocument) = await GetLspWorkspaceAndDocumentAsync(secondDocumentUri, testLspServer).ConfigureAwait(false);
         AssertEx.NotNull(secondDocument);
         Assert.Equal("Two is now three!", (await secondDocument.GetTextAsync()).ToString());
-        Assert.NotEqual(testLspServer.TestWorkspace.CurrentSolution.GetDocument(secondDocument.Id), secondDocument);
+
+        if (mutatingLspWorkspace)
+        {
+            // In the mutating case, opening the second doc pushes its changes through to the underlying workspace.  So
+            // the documents will be the same.
+            Assert.Equal(testLspServer.TestWorkspace.CurrentSolution.GetDocument(secondDocument.Id), secondDocument);
+        }
+        else
+        {
+            Assert.NotEqual(testLspServer.TestWorkspace.CurrentSolution.GetDocument(secondDocument.Id), secondDocument);
```

Comment:
Can something else be asserted, like you asserted state was the same in the previous test?